### PR TITLE
hypershift/openstack: create new workflow

### PR DESCRIPTION
**Pull request by EmilienM**
In the first implementation of e2e testing against Hypershift on
OpenStack, we created a chain that overrided the `test` stanza. But this
was not enough because we need to run steps before and after a cluster
has been created, tested and destroyed.

We have steps in pre that create what's needed for Ingress: a floating
IP and a DNS record.
We have steps in post to cleanup what was created in pre.

At this point, a new workflow was needed.

### CODE REVIEW

**Comment by EmilienM** [comment]
This worked fine.
/pj-rehearse ack

Now I'm waiting for feedback.

File: ci-operator/step-registry/hypershift/openstack/e2e/hypershift-openstack-e2e-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+workflow:
+  as: hypershift-openstack-e2e
+  documentation: |-
+    This workflow is a fork from `hypershift-aws-e2e-nested-workflow`.
+    It's deploying a management cluster on AWS and then prepare
```
Comment (mandre) [inline ci-operator/step-registry/hypershift/openstack/e2e/hypershift-openstack-e2e-workflow.yaml:5]: The workflow is doing a more than what is described here (we only document the `pre` step). Perhaps we could expand a bit and explain what the `test` step is doing at least.

Comment (EmilienM) [reply ci-operator/step-registry/hypershift/openstack/e2e/hypershift-openstack-e2e-workflow.yaml]: done

**Comment by EmilienM** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-openstack
the job should be green now

**Review by mandre — COMMENTED** [review]
/lgtm

**Comment by EmilienM** [comment]
/pj-rehearse ack
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/55246/rehearse-55246-pull-ci-openshift-hypershift-main-e2e-openstack/1821494634212233216 passed

**Pull request merged**